### PR TITLE
Make AccessPolicy::StartsOn and ExpiresOn nullable values.

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp [hunk 1]
```diff
@@ -5008,18 +5008,26 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
         {
           writer.Write(_internal::XmlNode{_internal::XmlNodeType::StartTag, "AccessPolicy"});
           writer.Write(_internal::XmlNode{_internal::XmlNodeType::StartTag, "Start"});
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:5010]: `<start>` tag and end tag should be moved into if clause

File: sdk/storage/azure-storage-files-shares/test/share_client_test.cpp [hunk 2]
```diff
@@ -213,6 +217,24 @@ namespace Azure { namespace Storage { namespace Test {
...
+  TEST_F(FileShareClientTest, ShareAccessPolicyNullable)
+  {
+    std::vector<Files::Shares::Models::SignedIdentifier> identifiers;
+    for (unsigned i = 0; i < 3; ++i)
+    {
+      Files::Shares::Models::SignedIdentifier identifier;
+      identifier.Id = RandomString(64);
+      identifier.Policy.Permission = "r";
+      identifiers.emplace_back(identifier);
+    }
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/test/share_client_test.cpp:229]: More test cases, such as `StartsOn` is null, `Expires` is null, both of them are null.

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-files-shares/test/share_client_test.cpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/test/share_client_test.cpp]: Added.

**Review by Jinming-Hu — APPROVED** [review]
